Amplify/Categories/DataStore/Model/Schema/ModelField+Association.swift

```diff
@@ -183,4 +183,25 @@ extension ModelField {
         return false
     }
 
+    public var embeddedType: Embedded.Type? {
+        switch type {
+        case .embedded(let type), .embeddedCollection(let type):
+            if let embeddedType = type as? Embedded.Type {
+                return embeddedType
+            }
+            return nil
+        default:
+            return nil
+        }
+    }
+
+    public var isEmbeddedType: Bool {
+        if case .embedded = type {
+            return true
+        }
+        if case .embeddedCollection = type {
+            return true
+        }
```

Comment:
you could potentially combine both using a `,` in the same `if` clause, or also use a `switch` where the default returns false.

Reply:
i think `if case .x = type, case .y = type` is checking for both case and will never be true. not completely sure if that is the case when i saw a test failed, so i switched to using the switch statement instead